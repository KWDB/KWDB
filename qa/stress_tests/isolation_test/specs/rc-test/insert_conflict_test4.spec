setup
{
  CREATE TABLE upsert (key text not null, payload text);
  CREATE UNIQUE INDEX ON upsert(key) INCLUDE (payload);
}

teardown
{
  DROP TABLE upsert;
}

session s1
setup       { BEGIN transaction isolation level read committed; }
step s1w    { INSERT INTO upsert(key, payload) VALUES('FooFoo', 'insert1') ON CONFLICT (key) DO UPDATE set key = EXCLUDED.key, payload = upsert.payload || ' updated by insert1'; }
step s1c    { COMMIT; }
step s1a    { ABORT; }

session s2
setup       { BEGIN transaction isolation level read committed; }
step s2w    { INSERT INTO upsert(key, payload) VALUES('FOOFOO', 'insert2') ON CONFLICT (key) DO UPDATE set key = EXCLUDED.key, payload = upsert.payload || ' updated by insert2'; }
step s2r    { SELECT * FROM upsert; }
step s2c    { COMMIT; }

permutation s1w s2w s1c s2r s2c
permutation s1w s2w s1a s2r s2c
