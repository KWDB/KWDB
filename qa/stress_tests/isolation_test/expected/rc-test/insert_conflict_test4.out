Parsed test spec with 2 sessions

starting permutation: s1w s2w s1c s2r s2c
step s1w: INSERT INTO upsert(key, payload) VALUES('FooFoo', 'insert1') ON CONFLICT (key) DO UPDATE set key = EXCLUDED.key, payload = upsert.payload || ' updated by insert1';  finished.
step s2w: INSERT INTO upsert(key, payload) VALUES('FOOFOO', 'insert2') ON CONFLICT (key) DO UPDATE set key = EXCLUDED.key, payload = upsert.payload || ' updated by insert2';  finished.
step s1c: COMMIT;  finished.
step s2r: SELECT * FROM upsert;  finished.
key   |payload
------+-------
FooFoo|insert1
FOOFOO|insert2
(2 rows)

step s2c: COMMIT;  finished.

starting permutation: s1w s2w s1a s2r s2c
step s1w: INSERT INTO upsert(key, payload) VALUES('FooFoo', 'insert1') ON CONFLICT (key) DO UPDATE set key = EXCLUDED.key, payload = upsert.payload || ' updated by insert1';  finished.
step s2w: INSERT INTO upsert(key, payload) VALUES('FOOFOO', 'insert2') ON CONFLICT (key) DO UPDATE set key = EXCLUDED.key, payload = upsert.payload || ' updated by insert2';  finished.
step s1a: ABORT;  finished.
step s2r: SELECT * FROM upsert;  finished.
key   |payload
------+-------
FOOFOO|insert2
(1 row)

step s2c: COMMIT;  finished.
